src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -303,15 +304,17 @@ public void registerUserForCourse(User user, Course course) {
      * Get the active students for these particular exercise ids
      *
      * @param exerciseIds the ids to get the active students for
+     * @param periodIndex the deviation from the current time
      * @return An Integer array containing active students for each index. An index corresponds to a week
      */
-    public Integer[] getActiveStudents(List<Long> exerciseIds) {
+    public Integer[] getActiveStudents(Set<Long> exerciseIds, Integer periodIndex) {
         ZonedDateTime now = ZonedDateTime.now();
         LocalDateTime localStartDate = now.toLocalDateTime().with(DayOfWeek.MONDAY);
         LocalDateTime localEndDate = now.toLocalDateTime().with(DayOfWeek.SUNDAY);
         ZoneId zone = now.getZone();
-        ZonedDateTime startDate = localStartDate.atZone(zone).minusWeeks(3).withHour(0).withMinute(0).withSecond(0).withNano(0);
-        ZonedDateTime endDate = localEndDate.atZone(zone).withHour(23).withMinute(59).withSecond(59);
+        ZonedDateTime startDate = localStartDate.atZone(zone).minusWeeks(3 + (4 * (-periodIndex))).withHour(0).withMinute(0).withSecond(0).withNano(0);
+        ZonedDateTime endDate = periodIndex != 0 ? localEndDate.atZone(zone).minusWeeks(4 * (-periodIndex)).withHour(23).withMinute(59).withSecond(59)
+                : localEndDate.atZone(zone).withHour(23).withMinute(59).withSecond(59);
```

Comment:
I'm still a bit confused here, especially the two calculations 3 + (4 * (-periodIndex)) and 4 * (-periodIndex). But I think this would make it a bit more intuitive:
```suggestion
        ZonedDateTime startDate = localStartDate.atZone(zone).minusWeeks(3 + (4 * (-periodIndex))).withHour(0).withMinute(0).withSecond(0).withNano(0);
        ZoneDateTime someFittingName = periodIndex != 0 ? localEndDate.atZone(zone).minusWeeks(4 * (-periodIndex)) : localEndDate;
        ZonedDateTime endDate = someFittingName.withHour(23).withMinute(59).withSecond(59);
```

Reply:
What should the `someFittingName` portray? In your case someFittingName could have two types, either ZonedDateTime if periodIndex != null or LocalDateTime if periodIndex == 0, if i am not mistaken. The reason why we distinguish for the endDate is that if we are looking at the current week, we take the end of the current day as the endDate. If we, however, look at another week, the endDate is the Sunday 23:59 of that particular week.

Reply:
You're right, I forgot the .atZone() in the second part of the ternary.
The someFittingName would be the end date without the changed time. I mostly tried to make the code a bit more readable...

Reply:
I added comments which describe what is happening here in e9883b7